src/serialization.rs

```diff
@@ -0,0 +1,61 @@
+use std::error::Error;
+
+use rustc_serialize::base64::{FromBase64, STANDARD, ToBase64};
+use serde::{Deserializer, Error as SerdeError, Serializer};
+use serde::de::Visitor;
+
+pub trait SerdeBlob: Sized {
+    fn deserialize_blob<D>(deserializer: &mut D) -> Result<Self, D::Error> where D: Deserializer;
+
+    fn serialize_blob<S>(&self, serializer: &mut S) -> Result<(), S::Error> where S: Serializer;
+}
+
+struct BlobVisitor;
+
+impl Visitor for BlobVisitor {
```

Comment:
These serde visitors will take some time for me to grasp.  I'll look more at documentation when I get a chance.